src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final String INDEX_OUT_OF_BOUND_MESSAGE =
+            "Index is less than 0 and more than last list index.";
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double CAPACITY_GROW_COEFFICIENT = 1.5;
+    private T[] values;
+    private int size;
+
+    ArrayList() {
+        values = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        checkSize();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return values[index];
+
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final T deletedValue = values[index];
```

Comment:
let's return `values[index] = null`, we need it (my fault)

Reply:
returned this code